monitoring/ping.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2019 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package monitoring
+
+import (
+	"context"
+	"log"
+
+	"github.com/gravitational/satellite/agent"
+	"github.com/gravitational/satellite/agent/health"
+	pb "github.com/gravitational/satellite/agent/proto/agentpb"
+	"github.com/gravitational/trace"
+	serf "github.com/hashicorp/serf/client"
+	ping "github.com/sparrc/go-ping"
+)
+
+const (
+	PingCheckerID = "ping-checker"
+)
+
+func NewPingChecker(serfRPCaddr string, role string) health.Checker {
+	return &PingChecker{
+		serfRPCAddr: serfRPCaddr,
+		role:        role,
+	}
+}
+
+// PingChecker is a checker that verifies if ping between Master Nodes and other
+// ones is below a specified threshold
+type PingChecker struct {
+	serfRPCAddr string
+	role        string
+}
+
+// Name returns the check name
+// Implements health.Checker
+func (c *PingChecker) Name() string {
+	return PingCheckerID
+}
+
+// Check verifies that all nodes' ping with Master Nodes is lower than the
+// desired threshold
+// Implements health.Checker
+func (c *PingChecker) Check(ctx context.Context, r health.Reporter) {
+	RttThreshold := 100     // ms
```

Comment:
I would make threshold a part of the checker configuration. Also 100ms seems too lenient, I'd consider everything above 15-20ms (maybe even 10ms) latency on an internal network a degraded performance.

Reply:
okay, that's definitely one of the feedback I wanted/needed. I was thinking of going that direction but didn't know where to go.
If that's okay I'll start with just using a more sensible (but conservative value) like 25ms and then work on making it a parameter later.

Reply:
Sounds good. I would also make sure we're logging these metrics so we can for example look at the collected data and adjust the value appropriately if needed.

Reply:
okay, value changed and FIXME added.